node/system_function.go

```diff
@@ -0,0 +1,67 @@
+package node
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/heroku/libhkbuildpack/layers"
+)
+
+type SystemFunction struct {
+	Path  string `toml:"path"`
+	Layer layers.Layer
+}
+
+func NewSystemFunction(l layers.Layer, path string) SystemFunction {
+	return SystemFunction{
+		Path:  path,
+		Layer: l,
+	}
+}
+
+func (f SystemFunction) Contribute() error {
+	f.Layer.Touch()
+
+	if err := f.Layer.WriteMetadata(f, layers.Launch); err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(f.Layer.Root, 0755); err != nil {
+		return err
+	}
+
+	filenames := []string{"index.js", "package.json", "package-lock.json"}
+	for _, filename := range filenames {
+		sourceFilename := filepath.Join(f.Path, "system/", filename)
```

Comment:
The slash appended to `system/` should be removed. 
`sourceFilename := filepath.Join(f.Path, "system", filename)`

Reply:
Good call. Done in https://github.com/heroku/node-function-buildpack/pull/23/commits/51c276c3d55229ab52aa16a4a34b69d0e7406296.